gadopt/energy_solver.py

```diff
@@ -145,16 +121,70 @@ def __init__(
 
         self.timestepper = timestepper
         self.T_old = Function(self.Q)
+
+        rho_cp = approximation.rhocp()
+        eq_terms = [
+            scal_eq.scalar_advection_term,
+            scal_eq.scalar_diffusion_term,
+            scal_eq.scalar_absorption_term,
+            scal_eq.scalar_source_term,
```

Comment:
```suggestion
            scalar_eq.advection_term,
            scalar_eq.diffusion_term,
            scalar_eq.absorption_term,
            scalar_eq.source_term,
```

In my opinion it's clearer to use the qualification to clarify it's scalar, rather than the term name.